### PR TITLE
Closes #362 Added libwebp package

### LINKED ISSUE #362: Add libwebp package
- [Official (?)](http://www.webmproject.org/)
- [GitHub mirror](https://github.com/webmproject/libwebp)

Notes:
- CMake based
- Used in [OpenCV](https://github.com/hunter-packages/opencv/tree/ab2d4be0ac4ef08c18ed534202d7881b736bc356/3rdparty/libwebp)
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/32284196-add-libwebp-package?utm_campaign=plugin&utm_content=tracker%2F1032790&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1032790&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by mchiasson**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **Yes**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/mchiasson/hunter/build/1.0.25
  * https://travis-ci.org/mchiasson/hunter/builds/352309209

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/203

* My change will resolve this "New package" request: https://github.com/ruslo/hunter/issues/362

---

### CODE REVIEW

**Comment by xsacha** [comment]
Thanks, will be good to add this to OpenCV.

By the way, I think you are meant to change those Yes/No to Yes or No and remove the parts that aren't relevant from the template.

**Comment by mchiasson** [comment]
@xsacha yup! OpenCV and also SDL_image, which also can make use of WebP. ;-)

Yes, I will change the Yes/No soon. It is my very first contribution to hunter, and I will make sure to do it all by your standards.

File: cmake/projects/WebP/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+        PACKAGE_NAME
+        WebP
+        VERSION
+        "0.6.1-p0"
+        URL
+        "https://github.com/mchiasson/libwebp/archive/v0.6.1-p0.tar.gz"
```
Comment (mchiasson) [inline cmake/projects/WebP/hunter.cmake]: FYI, this url is wrong, and will need to be changed to `hunter-package/libweb` 

https://github.com/hunter-packages/libwebp/pull/1

**Comment by mchiasson** [comment]
Note to self:  Missing `hunter/docs/packages/pkg/WebP.rst`

**Comment by ruslo** [comment]
> Here is the links to the Travis/AppVeyor with status "All passed"

I don't see any job with status "Passed", they all failed.

**Comment by mchiasson** [comment]
sorry @ruslo I'm still trying to figure out how to setup my travis-ci and AppVeyor here.. something is misconfigured on my end.

File: cmake/projects/WebP/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    WEBP_BUILD_GIF2WEBP=ON
+    WEBP_BUILD_IMG2WEBP=ON
+    WEBP_BUILD_WEBPINFO=ON
+    WEBP_BUILD_WEBP_JS=OFF
+    WEBP_ENABLE_NEAR_LOSSLESS=ON
+    WEBP_EXPERIMENTAL_FEATURES=OFF
+    WEBP_ENABLE_SWAP_16BIT_CSP=OFF
+)
+
+hunter_configuration_types(WebP CONFIGURATION_TYPES Release)
```
[the remark this answers is not in the document]
Comment (mchiasson) [reply cmake/projects/WebP/hunter.cmake]: you're right. removed.

File: examples/WebP/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+include("../common.cmake")
+
+project(download-webp)
+
+# DOCUMENTATION_START {
+hunter_add_package(WebP)
+find_package(WebP CONFIG REQUIRED)
+add_executable(main main.cpp)
+target_link_libraries(main WebP::webp)
+# DOCUMENTATION_END }
```
[the remark this answers is not in the document]
Comment (mchiasson) [reply examples/WebP/CMakeLists.txt]: done.

File: cmake/configs/default.cmake [hunk 4]
```diff
@@ -272,7 +272,7 @@ hunter_config(odb-sqlite VERSION 2.4.0)
 hunter_config(ogles_gpgpu VERSION 0.2.8)
 hunter_config(onmt VERSION 0.4.1-p2)
 hunter_config(openddlparser VERSION 0.1.0-p2)
-hunter_config(opentracing-cpp VERSION 1.1.0)
+hunter_config(opentracing-cpp VERSION 1.3.0)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: Doesn't belong here, I guess you should rebase your change.

Comment (mchiasson) [reply cmake/configs/default.cmake]: woops! good catch. not sure how this one got in there

File: cmake/configs/default.cmake [hunk 5]
```diff
@@ -393,3 +393,5 @@ hunter_config(zookeeper VERSION 3.4.9-p2)
 hunter_config(tacopie VERSION 2.4.0-h1)
 hunter_config(cpp_redis VERSION 3.5.0-h1)
 hunter_config(IF97 VERSION 2.1.2)
+hunter_config(WebP VERSION 0.6.1-p0)
```
[the remark this answers is not in the document]
Comment (mchiasson) [reply cmake/configs/default.cmake]: done

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.3421
* https://travis-ci.org/ingenue/hunter/builds/352327079?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.3425
* https://travis-ci.org/ingenue/hunter/builds/352346134?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.20.17

**Pull request merged**